core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/delete/DeleteKeystoresHandler.java

```diff
@@ -0,0 +1,229 @@
+
+  public DeleteKeystoresHandler(
+      final ObjectMapper objectMapper,
+      final Path keystorePath,
+      final Optional<SlashingProtection> slashingProtection,
+      final ArtifactSignerProvider signerProvider) {
+    this.objectMapper = objectMapper;
+    this.keystorePath = keystorePath;
+    this.slashingProtection = slashingProtection;
+    this.signerProvider = signerProvider;
+  }
+
+  @Override
+  public void handle(RoutingContext context) {
+    // API spec - https://github.com/ethereum/keymanager-APIs/tree/master/flows#delete
+    final RequestParameters params = context.get("parsedParameters");
+    final DeleteKeystoresRequestBody parsedBody;
+    try {
+      parsedBody = parseRequestBody(params);
+    } catch (final IllegalArgumentException | JsonProcessingException e) {
+      handleInvalidRequest(context, e);
+      return;
+    }
+
+    // normalize incoming keys to delete
+    final List<String> pubkeysToDelete =
+        parsedBody.getPubkeys().stream()
+            .map(IdentifierUtils::normaliseIdentifier)
+            .collect(Collectors.toList());
+
+    // load active keys
+    final Set<String> activePubkeys =
+        signerProvider.availableIdentifiers().stream()
+            .map(IdentifierUtils::normaliseIdentifier)
+            .collect(Collectors.toSet());
+
+    final List<DeleteKeystoreResult> results = new ArrayList<>();
+    final List<String> keysToExport = new ArrayList<>();
+    // process each incoming key individually
+    for (String pubkey : pubkeysToDelete) {
+      try {
+        final boolean isActive = activePubkeys.contains(pubkey);
+        final Optional<Path> keystoreConfigFile = findKeystoreConfigFile(pubkey);
+
+        // check that key is active
+        if (!isActive) {
+          // if not active, check if we ever had this key registered in the slashing DB
+          final boolean wasRegistered =
+              slashingProtection
+                  .map(protection -> protection.isRegisteredValidator(Bytes.fromHexString(pubkey)))
+                  .orElse(false);
+          // if it was registered previously, return not_active and add to list of keys to export,
+          // otherwise not_found
+          if (wasRegistered) {
+            keysToExport.add(pubkey);
+            results.add(new DeleteKeystoreResult(DeleteKeystoreStatus.NOT_ACTIVE, ""));
+          } else {
+            results.add(new DeleteKeystoreResult(DeleteKeystoreStatus.NOT_FOUND, ""));
+          }
+          continue;
+        }
+        // Remove active key from memory first, will stop any further signing with this key
+        signerProvider.removeSigner(pubkey).get();
+        // Then, delete the corresponding keystore file
+        if (keystoreConfigFile.isPresent()) {
+          Files.delete(keystoreConfigFile.get());
+        }
+        // finally, add result response
+        keysToExport.add(pubkey);
+        results.add(new DeleteKeystoreResult(DeleteKeystoreStatus.DELETED, ""));
```

Comment:
If the keystore is read-only then we should be returning an error as it hasn't been deleted